pkg/services/scm_provider/gitlab.go

```diff
@@ -112,10 +113,34 @@ func (g *GitlabProvider) RepoHasPath(_ context.Context, repo *Repository, path s
 	if err != nil {
 		return false, err
 	}
-	if resp.TotalItems == 0 {
-		return false, nil
+	// if it's a directory, we're done
+	if resp.TotalItems > 0 {
+		return true, nil
 	}
-	return true, nil
+
+	// let's try if path is a file
+	dir := pathpkg.Dir(path)
+	options := gitlab.ListTreeOptions{
+		Path: &dir,
+		Ref:  &repo.Branch,
+	}
+	for {
+		treeNode, resp, err := g.client.Repositories.ListTree(p.ID, &options)
```

Comment:
can we include one above this listTree call into a loop.

Reply:
Not sure if I understand you right: Do you like me to merge the duplicated `listTree` call into a loop?

If yes, I can't see how this will improve readability or maintainability of this function. Specially since the result of this two calls are differently interpreted:
1. if it's a directory any none empty list is ok
2. if it's a file we have to iterate over all pages to check if the directory contains the requested file

Reply:
Yes i feel it's a duplicated code and hence should be merged.

since it's a function, function returns whether path exists or not, 

on first call if it's directory  and in loop u can compare total items and break and return from function,

Currently same call is done twice

I feel implementing in  for loop with comments will improve readability.

Reply:
And thanks for addressing it,

I would request you to move this pr into Argo-cd repository as application set is merged there and future relase will happen along with argo-cd